src/nifc/gentypes.nim

```diff
@@ -159,22 +159,42 @@ proc genFieldPragmas(c: var GeneratedCode; types: TypeGraph; n: NodePos; bits: v
   else:
     error c.m, "expected field pragmas but got: ", types, n
 
+proc getIntQualify(types: TypeGraph; t: TypeId): string =
+  case types[t].kind
+  of RoC:
+    result = "const "
+  of AtomicC, Empty:
+    # TODO: implements it
+    result = ""
+  else:
+    raiseAssert "unreachable: " & $types[t].kind
+
 proc genType(c: var GeneratedCode; types: TypeGraph; t: TypeId; name = "") =
   template maybeAddName =
     if name != "":
       c.add Space
       c.add name
 
-  template atom(s: string) =
+  template atom(s: string, qualifier: string = "") =
+    c.add qualifier
     c.add s
     maybeAddName()
   case types[t].kind
   of VoidC: atom "void"
-  of IntC: atom "NI" & types.integralBits(t)
-  of UIntC: atom "NU" & types.integralBits(t)
-  of FloatC: atom "NF" & types.integralBits(t)
-  of BoolC: atom "NB8"
-  of CharC: atom "NC" & types.integralBits(t)
+  of IntC:
+    let (bits, qualifier) = sons2(types, t)
```

Comment:
You need something like `trySons2` here (which doesn't exist yet). Or maybe `countChildren check + ithSon`.